common/services/settings-validate.ts

```diff
@@ -0,0 +1,85 @@
+// Utils
```

Comment:
We need to discuss this service

- The name does not make much sense in English. `settings-validator` or `settings-validation` makes sense, but `settings-validate` doesn't.
- A function or method which responsibility is to check or validate something, usually returns a boolean. Most of the functions defined here not only return not boolean values, but return `undefined`. This is not only weird, it's also dangerous and prone to bugs. I understand that these strings are used as messages to be shown in the input for feedback. In this case, the following data structure could be used:

```ts
interface SettingValidation {
  isValid: boolean,
  reason: string
}

// ------------ //

function isString(value: any): SettingValidation {};

// ------------ //

const validationResult = isString(13);
if (validationResult.isValid) {
  input.validationError = validationResult.reason;
}
```

Additionally, this service could be a class:

```ts

class SettingsValidator {

  static isString() {};
  static isEmptyString() {};
  static stringHasMultipleLines() {};
  ...
  static isBoolean() {};
}

// ------------ //

const validationResult = SettingValidator.isString(13);
```

Finally, function declarations are recommended over function expression, following our **[Styleguide](https://github.com/wazuh/wazuh-kibana-app/blob/master/STYLEGUIDE.md#function-definitions)**.

Reply:
> We need to discuss this service
> 
> * The name does not make much sense in English. `settings-validator` or `settings-validation` makes sense, but `settings-validate` doesn't.
> * A function or method which responsibility is to check or validate something, usually returns a boolean. Most of the functions defined here not only return not boolean values, but return `undefined`. This is not only weird, it's also dangerous and prone to bugs. I understand that these strings are used as messages to be shown in the input for feedback. In this case, the following data structure could be used:
> 
> ```ts
> interface SettingValidation {
>   isValid: boolean,
>   reason: string
> }
> 
> // ------------ //
> 
> function isString(value: any): SettingValidation {};
> 
> // ------------ //
> 
> const validationResult = isString(13);
> if (validationResult.isValid) {
>   input.validationError = validationResult.reason;
> }
> ```

> Finally, function declarations are recommended over function expression, following our **[Styleguide](https://github.com/wazuh/wazuh-kibana-app/blob/master/STYLEGUIDE.md#function-definitions)**.

The validation functions are used in the server and frontend.

Their definition (input/output) is based on the `validate` parameter of `@kbn/config-schema` package. They return a `string` or `undefined`. The current definition allows us to use it directly as the `validate` parameter without wrappers. More information at https://github.com/elastic/kibana/tree/v7.10.2/packages/kbn-config-schema#custom-validation. This means that returning an object as it is suggested:
```ts
interface SettingValidation {
  isValid: boolean,
  reason: string
}
```
will not be compatible if we use that function as the `validate` property so we will need to wrap it to manage the output returning a `string` or `undefined`.

I think the approach for the validation is simple depending on the output type:
- `string`: the reason for the validation error.
- `undefined`: no validation error.

In the frontend, these functions are used to validate the input in the form of `Settings/Configuration`. If these functions return a string, then we can:
- display the field is invalid
- display the reason for the validation error

One value is used for 2 things.

> Additionally, this service could be a class:
> 
> ```ts
> class SettingsValidator {
> 
>   static isString() {};
>   static isEmptyString() {};
>   static stringHasMultipleLines() {};
>   ...
>   static isBoolean() {};
> }
> 
> // ------------ //
> 
> const validationResult = SettingValidator.isString(13);
> ```

We should discuss it and define it in the style guide. Functionally and in this case, functions are the same as classes with static methods.